libtalek/client.go

```diff
@@ -95,30 +96,42 @@ func (c *Client) MaxLength() uint64 {
 func (c *Client) Publish(handle *Topic, data []byte) error {
 	config := c.config.Load().(ClientConfig)
 
-	if len(data) > int(config.DataSize)-PublishingOverhead {
-		return errors.New("message too long")
-	} else if len(data) < int(config.DataSize)-PublishingOverhead {
+	if len(data) > 65535 {
+		return errors.New("message is too long")
+	}
+
+	// First word is prepended as length of data:
+	prefixed := make([]byte, 4, 4+len(data))
+	binary.LittleEndian.PutUint32(prefixed[0:4], uint32(len(data)))
+	data = append(prefixed, data...)
+
+	left := len(data)
+	for left > 0 {
```

Comment:
I think the issue with doing it this way is that you're assuming the client gets every single message. In order to support missing a bunch of messages and coming back later, we need to prefix every message with 1. the item length, 2. how many partial messages exist for this message, and 3. which partial message this is. That way we can go offline, come back, and at least know what to ignore to wait for the next full message